Update Readme (#1)
DevExpress-Examples/asp-net-web-forms-grid-load-file-from-client: Readme.md

[![](https://img.shields.io/badge/Open_in_DevExpress_Support_Center-FF7200?style=flat-square&logo=DevExpress&logoColor=white)](https://supportcenter.devexpress.com/ticket/details/E2577)
[![](https://img.shields.io/badge/📖_How_to_use_DevExpress_Examples-e9f6fc?style=flat-square)](https://docs.devexpress.com/GeneralInformation/403183)
<!-- default badges end -->

# Grid View for ASP.NET Web Forms - How to load a file from the client side
<!-- run online -->
**[[Run Online]](https://codecentral.devexpress.com/e2577/)**
<!-- run online end -->

> This example shows how to load a file when a user clicks a custom command button in Grid. 

Reviewer: ```suggestion
This example shows how to load a file when a user clicks a custom command button in the Grid. 
```

![](grid.png)

> When a user clicks a button, the corresponding file name, which equals to the **Employee ID**, is added to the `window.location.href` property as a parameter. On the server side, the file with the specified name is send to the client in response.

Reviewer: ```suggestion
When a user clicks a button, the file with the corresponding **Employee ID**, is added to the `window.location.href` property as a parameter. On the server side, the file with the specified name is sent to the client in response.
```

```cs
protected void Page_Load(object sender, EventArgs e) {
    if (Request["ID"] != null)
        SendFile(Request["ID"]);
// ...
```

This example demonstrates how to specify the uploaded file name in two ways:

> * Specify the name completely on the client side.

Reviewer: ```suggestion
* Specify the name on the client.
```

> * Send a callback to get the name from the server side.

Reviewer: ```suggestion
* Send a callback to get the name from the server.
```

For demonstration purposes, in both approaches it is prohibited to download the **9.png** file.

> ## Specify the file name completely on the client side.

Author: ```suggestion
## Specify the file name on the client.
```

Handle the [CustomButtonClick](https://docs.devexpress.com/AspNet/js-ASPxClientGridView.CustomButtonClick) event to set the `window.location.href` property.

```js
function DoClientRedirect(value) {
    window.location.href = 'Default.aspx?ID=' + value;
}
```

> ## Send a callback to get the file name from the server side.

Reviewer: ```suggestion
## Send a callback to get the file name from the server.
```
